staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -126,6 +136,8 @@ func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attr
 			if callErr, ok := err.(*webhookutil.ErrCallingWebhook); ok {
 				if ignoreClientCallFailures {
 					klog.Warningf("Failed calling webhook, failing open %v: %v", hook.Name, callErr)
+
+					versionedAttr.Attributes.AddAnnotation(ValidationAuditAnnotationPrefix  + ValidationAuditAnnotationFailedOpenKeySuffix, "true")
```

Comment:
There can be multiple fail-open webhook calls for one request and this doesn't distinguish them

Reply:
Sure will append the hook's name in the key.

Reply:
Also there are some things to watch out: 

1. apiserver validates the [annotation keys](https://github.com/kubernetes/kubernetes/blob/e45a598a5ca3e722eedc2d491b0c82fd7e77dc24/staging/src/k8s.io/apiserver/pkg/admission/attributes.go#L207) with some length limit (253-char prefix and 63-char name), while [webhook name](https://github.com/kubernetes/kubernetes/blob/e45a598a5ca3e722eedc2d491b0c82fd7e77dc24/pkg/apis/admissionregistration/validation/validation.go#L213) has a 253-char limit
2. webhook names are not validated for [uniqueness](https://github.com/kubernetes/kubernetes/blob/e45a598a5ca3e722eedc2d491b0c82fd7e77dc24/staging/src/k8s.io/apiserver/pkg/admission/configuration/validating_webhook_manager.go#L87-L95) (UIDs are), and apiserver doesn't allow [overwrite](https://github.com/kubernetes/kubernetes/blob/e45a598a5ca3e722eedc2d491b0c82fd7e77dc24/staging/src/k8s.io/apiserver/pkg/admission/attributes.go#L155-L157) existing annotation key

I think in reality it should be rare to have webhooks with super long names, or even with duplicated names. But just watch out that the annotation could fail silently if that happened, given that the code ignores the returned error

Reply:
Thanks for the watch outs and pointers, really helpful.

I decided to prepend the hook's name in the front, making it `<hook-name>.failure.webhook.admission.k8s.io`, which would resolve the distinguish-ness issue.  I also considered make separate annotation key/value pairs for the hook names, e.g. `failure.webhook.admission.k8s.io/failed-open-hook-{1..N}` -> `<hook-name>`, but it might be too boilerplate. Let me know what you think.

Also I added a klog.Warning if the annotation was not added successfully (e.g. name was too long due to the reasons you mentioned above)